jujugui/static/gui/src/app/components/login/login.js

```diff
@@ -23,14 +23,28 @@ YUI.add('login-component', function() {
   juju.components.Login = React.createClass({
 
     propTypes: {
+      controllerIsConnected: React.PropTypes.func.isRequired,
       errorMessage: React.PropTypes.string,
+      gisf: React.PropTypes.bool.isRequired,
       isLegacyJuju: React.PropTypes.bool.isRequired,
+      localStorageGet: React.PropTypes.func,
       loginToAPIs: React.PropTypes.func.isRequired,
-      loginToController: React.PropTypes.func.isRequired
+      loginToController: React.PropTypes.func.isRequired,
+      sendPost: React.PropTypes.func
     },
 
     componentDidMount: function () {
       this.refs.username.focus();
+      if (this.props.gisf) {
+        const bounce = () => {
+          if (this.props.controllerIsConnected) {
+            this.refs.USSOLoginLink.handleLogin();
```

Comment:
It's possible that we don't render the usso link in certain situations, could you wrap this with a conditional checking that the USSO link exists and that the handleLogin one does as well?